### PR TITLE
ConfigurationContainerScope: Add property name extensions for resolva…

### LINKED ISSUE #26601: New Configuration shortcuts: Add RegisteringDomainObjectDelegateProvider/RegisteringDomainObjectDelegateProviderWithAction overload
### Expected Behavior

```kotlin
val myConfig: NamedDomainObjectProvider<ConsumableConfiguration> by configurations.consumable {
    
}

val myConfig: NamedDomainObjectProvider<ConsumableConfiguration> by configurations.consumable

// similar to 
val myConfig: NamedDomainObjectProvider<Configuration> by configurations.registering {

}
val myConfig: NamedDomainObjectProvider<Configuration> by configurations.registering
```

### Current Behavior (optional)

```kotlin
val myConfig: NamedDomainObjectProvider<ConsumableConfiguration> by configurations.consumable("myConfig") {
    
}
val myConfig: NamedDomainObjectProvider<ConsumableConfiguration> by configurations.consumable("myConfig")
```

### Context

I currently use the `configurations.registering` provider to only specify the name of the configuration once by using the variable name. The new api does not have this overload, so during migration to the new configurations types, I need to specify the configuration names twice.

### PR DESCRIPTION
**Pull request by hfhbd**
…ble, consumable and dependencyScope

Fixes #26601

### Contributor Checklist
- [X] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [X] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [X] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [X] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [X] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [X] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [X] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [X] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: platforms/core-configuration/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/DependencyManagementIntegrationTest.kt [hunk 1]
```diff
@@ -131,4 +132,55 @@ class DependencyManagementIntegrationTest : AbstractKotlinIntegrationTest() {
...
+
+        withFile("repo/in-block/accessor-1.0.jar")
+
+        withBuildScript(
+            """
+            val foo by configurations.dependencyScope
+            val bar by configurations.dependencyScope { }
+            
+            dependencies {
+                foo.name("in-block:accessor:1.0")
```
Comment (hfhbd) [inline platforms/core-configuration/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/DependencyManagementIntegrationTest.kt:148]: See #26602

File: platforms/core-configuration/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/ConfigurationContainerExtensions.kt [hunk 2]
```diff
@@ -84,3 +86,114 @@ private constructor(
...
+
+
+/**
+ * Provides access to the [NamedDomainObjectProvider] for the element of the given
+ * property name from the configuration container via a delegated property.
+ *
+ * @since 8.5
+ */
+@Incubating
+class ConfigurationProvider<C : Configuration> internal constructor(private val createProvider: (String) -> NamedDomainObjectProvider<C>) {
```
Comment (hfhbd) [inline platforms/core-configuration/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/ConfigurationContainerExtensions.kt:191]: Instead exposing this class to the ABI, the kotlin std provides [PropertyDelegateProvider](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.properties/-property-delegate-provider/), but this interface is not allowed in Gradle public ABI (yet), while it is still a valid Kotlin std class/interface. Any chance to relaxe the ABI validator and allow Kotlin std types in the Gradle Kotlin DSL only?

**Comment by cobexer** [comment]
Thank you for your contribution!

This PR has a valid DCO and tests. The relevant team for this area will confirm the design of the implementation choices.

**Comment by big-guy** [comment]
At the moment, we don't want to add delegated properties for these new methods.  This makes the Kotlin and Groovy DSLs more different.

Thanks for your contributions.

**Comment by hfhbd** [comment]
How is this different to the existing tasks or DomainCollection delegated properties only provided for the Kotlin DSL without a groovy counterpart?

**Comment by jvandort** [comment]
It's not different. We're not sure the existing ones were the best idea

**Pull request closed without merging**